airflow/jobs/scheduler_job.py

```diff
@@ -1647,6 +1649,13 @@ def _process_and_execute_tasks(self, simple_dag_bag):
                                                   State.NONE)
         self._execute_task_instances(simple_dag_bag)
 
+    def _collect_metrics(self) -> None:
```

Comment:
```suggestion
    def _emit_pool_metrics(self) -> None:
```

as we do more than collect them here.

Reply:
Oh, and we should pass in the session, and pass that down to slot_stats.